runtime/metricsview/query.go

```diff
@@ -26,6 +26,7 @@ type Query struct {
 	TimeZone            string      `mapstructure:"time_zone"`
 	UseDisplayNames     bool        `mapstructure:"use_display_names"`
 	Rows                bool        `mapstructure:"rows"`
+	TimeColumn          string      `mapstructure:"time_column"` // optional time column to use for time-based operations, if not specified, the default time column in the metrics view is used
```

Comment:
First, can we avoid using the word "column" since that refers to underlying table columns, which doesn't match dimension/field, which can also have expressions?

Second, instead of passing the time dimension as a root property, would it be possible to take it in the individual constructs instead, specifically in `Query.Dimensions[*].Compute.Dimension` (already exists) and `TimeRange.Dimension` (would need to be added)?

That would enable e.g. aggregating by two different time dimensions, or filtering by a different time dimension than you are aggregating by. I'm not sure, but it _might_ also make the code a bit simpler.

Only thing is for comparisons, I think we'd have to require that `Query.TimeRange.Dimension == Query.ComparisonTimeRange.Dimension`.

Reply:
> instead of passing the time dimension as a root property, would it be possible to take it in the individual constructs instead, specifically in Query.Dimensions[*].Compute.Dimension (already exists) and TimeRange.Dimension (would need to be added)?

I understand that a `Dimension` field can be added to `TimeRange` that specifies which time dim to use for filtering instead of passing `time_dimension` as a separate argument in the API. I don't understand what you mean by passing time dim in `Query.Dimensions[*].Compute.Dimension` is it so that it can be rolled up and used in the group by? thats already possible even when using more than time dims.

Reply:
>  I don't understand what you mean by passing time dim in Query.Dimensions[*].Compute.Dimension
Currently it works like this:
```
{"dimensions": {"name": "__time__hour", "compute":{"time_floor": {"grain": "hour", "dimension":"__time"}}}
```

Notice how the first `name` field is just an alias, and the actual time dimension is given in the nested `query.dimensions[*].compute.time_floor.dimension` property?

So what I mean is, the time dimension name is explicitly given here, so a base `query.time_column` can perhaps be avoided.

This also allows a query to filter the time range by another time dimension than they are doing the grouping by.